powerline/listers/i3wm.py

```diff
@@ -0,0 +1,78 @@
+# vim:fileencoding=utf-8:noet
+from __future__ import (unicode_literals, division, absolute_import, print_function)
+
+import re
+
+from powerline.theme import requires_segment_info
+from powerline.lib.shell import run_cmd
+
+
+conn = None
+
+
+@requires_segment_info
+def output_lister(pl, segment_info):
+	'''List all outputs in segment_info format'''
+
+	outputs = re.findall(
+		'^([0-9A-Za-z-]+) connected',
+		run_cmd(pl, ['xrandr', '-q']),
+		re.MULTILINE
+	)
+	return (
+		(
+			{
+				'output': output
+			},
+			{}
+		)
+		for output in outputs
+	)
+
+
+@requires_segment_info
+def workspace_lister(pl, segment_info, only_show=None, output=None):
+	'''List all workspaces in segment_info format
+
+	Sets the segment info values of ``workspace`` and ``output`` to the name of 
+	the i3 workspace and the ``xrandr`` output respectively and the keys
+	``"visible"``, ``"urgent"`` and ``"focused`` to a boolean indicating these
+	states.
+
+	:param list only_show:
+		Specifies which workspaces to list. Valid entries are ``"visible"``, 
+		``"urgent"`` and ``"focused"``. If omitted or ``null`` all workspaces 
+		are listed.
+
+	:param str output:
+		May be set to the name of an X output. If specified, only workspaces 
+		on that output are listed. Overrides automatic output detection by 
+		the lemonbar renderer and bindings.
+	'''
+
+	global conn
+	if not conn:
+		try:
+			import i3ipc
+		except ImportError:
+			import i3 as conn
+		else:
+			conn = i3ipc.Connection()
+
+	output = output or segment_info.get('output')
+
+	return (
+		(
+			{
+				'workspace': w['name'],
+				'output': w['output'],
+				'visible': w['visible'],
+				'urgent': w['urgent'],
+				'focused': w['focused']
+			},
+			{}
+		)
+		for w in conn.get_workspaces()
+			if (not only_show or any(w[typ] for typ in only_show))
+			and (not output or w['output'] == output)
```

Comment:
This looks very strange. `if` should be indented with `for`. `and` should not be on the next line at all *or* the whole `if` conditional should be inside parenthesis.